google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryResultSetImpl.java

```diff
@@ -0,0 +1,554 @@
+        return curVal == null ? 0.0d : ((BigDecimal) curVal).doubleValue();
+      }
+    }
+
+    @Override
+    public double getDouble(int columnIndex) throws SQLException {
+      if (cursor == null) {
+        return 0d; // the column value; if the value is SQL NULL, the value returned is 0 as per
+        // java.sql.ResultSet definition
+      } else if (cursor instanceof FieldValueList) {
+        FieldValue fieldValue = ((FieldValueList) cursor).get(columnIndex);
+        return fieldValue.getValue() == null ? 0d : fieldValue.getNumericValue().doubleValue();
+      } else { // Data received from Read API (Arrow)
+        return getDouble(schemaFieldList.get(columnIndex).getName());
+      }
+    }
+
+    @Override
+    public BigDecimal getBigDecimal(String fieldName) throws SQLException {
+      if (fieldName == null) {
+        throw new SQLException("fieldName can't be null");
+      }
+      if (cursor == null) {
+        return null; // the column value (full precision); if the value is SQL NULL, the value
+        // returned is null in the Java programming language. as per
+        // java.sql.ResultSet definition
+      } else if (cursor instanceof FieldValueList) {
+        FieldValue fieldValue = ((FieldValueList) cursor).get(fieldName);
+        return fieldValue.getValue() == null
+            ? null
+            : BigDecimal.valueOf(fieldValue.getNumericValue().doubleValue());
+      } else { // Data received from Read API (Arrow)
+        return BigDecimal.valueOf(getDouble(fieldName));
+      }
+    }
+
+    @Override
+    public BigDecimal getBigDecimal(int columnIndex) throws SQLException {
+      if (cursor == null) {
+        return null; // the column value (full precision); if the value is SQL NULL, the value
+        // returned is null in the Java programming language. as per
+        // java.sql.ResultSet definition
+      } else if (cursor instanceof FieldValueList) {
+        FieldValue fieldValue = ((FieldValueList) cursor).get(columnIndex);
+        return fieldValue.getValue() == null
+            ? null
+            : BigDecimal.valueOf(fieldValue.getNumericValue().doubleValue());
+      } else { // Data received from Read API (Arrow)
+        return getBigDecimal(schemaFieldList.get(columnIndex).getName());
+      }
+    }
+
+    @Override
+    public boolean getBoolean(String fieldName) throws SQLException {
+      if (fieldName == null) {
+        throw new SQLException("fieldName can't be null");
+      }
+      if (cursor == null) {
+        return false; //  if the value is SQL NULL, the value returned is false
+      } else if (cursor instanceof FieldValueList) {
+        FieldValue fieldValue = ((FieldValueList) cursor).get(fieldName);
+        return fieldValue.getValue() != null && fieldValue.getBooleanValue();
+      } else { // Data received from Read API (Arrow)
+        Tuple<Map<String, Object>, Boolean> curTuple = (Tuple<Map<String, Object>, Boolean>) cursor;
```

Comment:
Yes, it can cause a ClassCastException if the value is not boolean. For example if someone calls getBoolean(intFiled) then they get the following exception:

```
Exception in thread "main" **java.lang.ClassCastException**: class **java.math.BigDecimal cannot be cast to class java.lang.Boolean** (java.math.BigDecimal and java.lang.Boolean are in module java.base of loader 'bootstrap')
	at com.google.cloud.bigquery.BigQueryResultSetImpl$ResultSetWrapper.getBoolean(BigQueryResultSetImpl.java:349)
	at localtest.ConnectionImplTest.main(ConnectionImplTest.java:68)

```
So, basically this transparently tell the developer that what has gone wrong. We can however wrap it as a `BigQuerySQLException` if you think that's a better approach.